src/native/libs/System.Security.Cryptography.Native.Apple/pal_swiftbindings.swift

```diff
@@ -350,3 +359,285 @@ public func AppleCryptoNative_HKDFDeriveKey(
+        return nil
+    }
+
+    switch hashAlgorithm {
+        case .md5:
+            pcbDigest.pointee = Int32(Insecure.MD5.byteCount)
+            let box = HashBox(Insecure.MD5(), algorithm: .md5)
+            return Unmanaged.passRetained(box).toOpaque()
+        case .sha1:
+            pcbDigest.pointee = Int32(Insecure.SHA1.byteCount)
+            let box = HashBox(Insecure.SHA1(), algorithm: .sha1)
+            return Unmanaged.passRetained(box).toOpaque()
+        case .sha256:
+            pcbDigest.pointee = Int32(SHA256.byteCount)
+            let box = HashBox(SHA256(), algorithm: .sha256)
+            return Unmanaged.passRetained(box).toOpaque()
+        case .sha384:
+            pcbDigest.pointee = Int32(SHA384.byteCount)
+            let box = HashBox(SHA384(), algorithm: .sha384)
+            return Unmanaged.passRetained(box).toOpaque()
+        case .sha512:
+            pcbDigest.pointee = Int32(SHA512.byteCount)
+            let box = HashBox(SHA512(), algorithm: .sha512)
+            return Unmanaged.passRetained(box).toOpaque()
+        default:
+            pcbDigest.pointee = 0
+            return nil
+    }
+}
+
+@_silgen_name("AppleCryptoNative_DigestUpdate")
+@available(iOS 13, tvOS 13, macOS 13, *)
+public func AppleCryptoNative_DigestUpdate(ctx: UnsafeMutableRawPointer?, pBuf: UnsafeMutableRawPointer?, cBuf: Int32) -> Int32 {
+    if cBuf == 0 {
+        return 1
+    }
+
+    guard let ctx, let pBuf, cBuf >= 0 else {
+        return -1
+    }
+
+    let box = Unmanaged<HashBox>.fromOpaque(ctx).takeUnretainedValue()
+    let source = Data(bytesNoCopy: pBuf, count: Int(cBuf), deallocator: Data.Deallocator.none)
+
+    switch box.algorithm {
+        case .md5:
+            var hash = box.value as! Insecure.MD5
+            hash.update(data: source)
+            box.value = hash
+            return 1
+        case .sha1:
+            var hash = box.value as! Insecure.SHA1
+            hash.update(data: source)
+            box.value = hash
+            return 1
+        case .sha256:
+            var hash = box.value as! SHA256
+            hash.update(data: source)
+            box.value = hash
+            return 1
+        case .sha384:
+            var hash = box.value as! SHA384
+            hash.update(data: source)
+            box.value = hash
+            return 1
+        case .sha512:
+            var hash = box.value as! SHA512
+            hash.update(data: source)
+            box.value = hash
+            return 1
+        default:
+            return -1
+    }
+}
+
+@_silgen_name("AppleCryptoNative_DigestReset")
+@available(iOS 13, tvOS 13, macOS 13, *)
+public func AppleCryptoNative_DigestReset(ctx: UnsafeMutableRawPointer?) -> Int32 {
+    guard let ctx else {
+        return -1
+    }
+
+    let box = Unmanaged<HashBox>.fromOpaque(ctx).takeUnretainedValue()
+
+    switch box.algorithm {
+        case .md5:
+            box.value = Insecure.MD5()
```

Comment:
I'm guessing that `init()` isn't really a function you can call, and that it's more like a constructor; so the type-unrolling here is therefore necessary.

Though it makes me wonder if you could replace the enum in the box with the uninvoked "SHA384", so this ends up as just `box.value = box.alg()`?